docker-app/qfieldcloud/core/models.py

```diff
@@ -2000,8 +2070,34 @@ def method(self):
         return self.content.get("method")
 
 
+class JobQuerySet(InheritanceQuerySet):
+    def for_user(self, user: User) -> models.QuerySet[Job]:
+        """Returns the jobs applicable to the user. If the user has assigned secrets, only jobs triggered by the user are returned.
+
+        Args:
+            user: The user to check for.
+
+        Returns:
+            The jobs for the user.
+        """
+        has_assigned_to_current_user_project_secrets = False
+        for project in user.projects.all():
+            if (
+                Secret.objects.for_user_and_project(user, project)  # type: ignore[attr-defined]
+                .filter(assigned_to__isnull=False)
+                .exists()
+            ):
+                has_assigned_to_current_user_project_secrets = True
```

Comment:
```suggestion
        for project in user.projects.all():
            if (
                Secret.objects.for_user_and_project(user, project)  # type: ignore[attr-defined]
                .filter(assigned_to__isnull=False)
                .exists()
            ):
                has_assigned_to_current_user_project_secrets = True
```

You have imperative code here (the `for` loop). This means:
1. first query to the database
2. then looping N times
3. each loop has a query to the database

No need to convince you more, we can make this in a single query. You already have warmed up with the querystrings :)

Reply:
Also if you for some reason still need to use `has_assigned_to_current_user_project_secrets`, I would try to shorten it a bit, or at least rename to `has_secrets_assigned_to_current_user_project`. What about `has_assigned_to_secrets`, and assume the reader knows `assigned_to` is a property is related to users?

Reply:
Not sure I'm following you.

Also, don't see changes in the gh suggestion.

Reply:
The two comments are unrelated, let us clarify over chat.

Reply:
Alright, thanks.

Would the following be acceptable:

```python
has_assigned_to_current_user_project_secrets = Secret.objects.for_user_and_projects(
    user=user,
    projects=user.projects.all()
).filter(
    assigned_to__isnull=False
).exists()
```

There is still this `user.projects.all()` call as a parameter though.